src/server/replica.cc

```diff
@@ -431,43 +432,40 @@ error_code Replica::InitiateDflySync() {
     shard_flows_[i].reset(new Replica(master_context_, i, &service_));
   }
 
-  SyncBlock sb{num_df_flows_};
+  // Allocate shared, because the error handler might outlive the function scope.
+  auto sync_block = std::make_shared<SyncBlock>();
 
-  AggregateError ec;
+  auto err_handler = [this, sync_block](const auto& ge) {
+    sync_block->Add(num_df_flows_);  // Unblock sync_block.
+    DefaultErrorHandler(ge);         // Close sockets to unblock flows.
+  };
+  if (cntx_.Switch(std::move(err_handler)))
+    return cntx_;
+
```

Comment:
I've added `Switch` to the context that atomically checks for error and exchanges the error handler